Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -470,7 +471,9 @@ public enum DefaultFunctionOpacityOptions {
       IncludeRuntimeOption,
       WarnContradictoryAssumptions,
       WarnRedundantAssumptions,
-      DefaultFunctionOpacity
+      DefaultFunctionOpacity,
+      JavaBackend.OuterPackage,
+      CsharpBackend.OuterNamespace
```

Comment:
I have trouble following. You think we MAY want to let something work in way X, so now we should already do X_part1 ? Why not wait until we know what we want? Right now `.doo` doesn't contain translated code or a derivative, so what sense is there in including translation options? If you change `.doo` so it contains more, than we can also add the options.

If you include `.doo` in something else that contains translated code, then you have no guarantee that the .doo file was generated with the same translation options as the generated code.

I would even go so far as to say translation options shouldn't be available when building a .doo file, so they will be ignored from your project file etc.

Reply:
That's totally fair. I'm just aiming to at least keep track of all the options that affect compilation output in a way that can break separate compilation, for when we do have the mechanism (and I'm working on that as part of adding standard libraries). Are you okay with putting these options in the `RegisterLibraryChecks` list with a comment for now?

I don't feel that strongly about it mind you. I should probably do another pass of reviewing all options anyway rather than trusting the existing lists.

Reply:
> I should probably do another pass of reviewing all options anyway rather than trusting the existing lists.

I'd prefer that if it's OK with you. "Design for today" was what my first programming mentor always said.

Reply:
Also I'm skeptical it makes sense to force translation options to be the same between different Dafny components that are integrating. There's also down-the-line compilation options, whatever you pass to other compilers, that might also influence something, but I feel that is outside of the scope of Dafny.

Dafny will ensure that Dafny's semantics are present when executing your program, but other things like the maximum stack size used when executing, are not Dafny's responsibility.

Reply:
Yup, not a problem. I'll take responsibility for the mechanism to track separate compilation compatibility as part of the standard library work.